greeter/README.md

```diff
@@ -1,21 +1,706 @@
-# Step 1
+# Write scalable and extensible Node.js applications using LoopBack 4
 
-- Create Message type file
-- Create a greeting controller
-- Create a mock greeter service
-- Bind the service to app
-- Try endpoint to print out hello world
+## What is LoopBack?
 
-# Step 2
+[LoopBack](https://loopback.io/) is a popular open source API and Micro-service framework
+in Node.js. Built on top of Express, it enables developers to quickly create REST and other forms of APIs or Microservices to interact with backend databases, services and other infrastructure using model driven design patterns. The command line tool makes the API creation easy and productive.
 
-- Mount greeter service component
-- Try endpoint with different language settings
-- Create a fr greeter
-- Register it to the app
-- Try endpoint again
+Rewritten in TypeScript from ground-up, LoopBack 4 core offers an Inversion of Control container with powerful Dependency Injection capabilities as the foundation to build large-scale Node.js projects with great flexibility and extensibility. LoopBack also integrates with cloud native health, metrics, and tracing to facilitate development of Micro--services for cloud deployment.
 
-# Step 3
+## Target Users
 
-- Add caching observer
-- Add caching interceptor
-- Add caching service
\ No newline at end of file
+- API developers
+- Platform developer + Extension developer
+
+## About This Workshop
+
+As your Node.js application becomes more complex, how do you increase its scalability with proven design patterns and best practices? How do you follow 12-factor principles to make your application flexible, extensible and composable? Let’s run a fun workshop to create a scalable and extensible application! While we could do this from scratch, we'll get there much faster and easier using a framework that does a lot of the hard work for us.
+
+In this workshop, you will be guided to scaffold a greeter app using the LoopBack 4 framework, and create a simple greeting API that prints out a welcome message such as "Hello, LoopBack", then enrich it with the idea of microservices without modifying its implementation in the next 4 steps:
+
+1. Creating a greeting endpoint that invokes service to generate the greeting message
+
+- we will show that how to simplify and break down your controller by injecting services.
```

Comment:
- maybe it's better to have the bullets indented?  similarly for the bullets below
- should we capitalize the `We` as the first character of the bullet?
- I'm thinking whether we should the detailed description here or move it as the summary.

Reply:
I come up the key concept here to give a high level idea of what we are going to show. And also add brief explanation for these concepts in each part/step. I can go either way~